framework/src/main/java/org/checkerframework/framework/util/element/ElementAnnotationUtil.java

```diff
@@ -451,7 +482,8 @@ private static AnnotatedTypeMirror getLocationTypeADT(
                     if (currentLocation.arg < innerType.getTypeArguments().size()) {
                         AnnotatedTypeMirror typeArg =
                                 innerType.getTypeArguments().get(currentLocation.arg);
-                        return getTypeAtLocation(typeArg, tailOfLocations);
+                        return getTypeAtLocation(
+                                typeArg, tailOfLocations, anno, isComponentTypeOfArray);
```

Comment:
Is this necessary in a type argument position? I thought only top-level annos are an issue?